prolog/st/st_tokens.pl

```diff
@@ -21,11 +21,35 @@
 
 st_tokens(Codes, Options, Tokens):-
     option(frontend(Frontend), Options),
-    must_be(oneof([simple, semblance]), Frontend),
-    phrase(tokens(Tmp1, Frontend), Codes),
+    (( Frontend =
+            syntax_tokens(
+                comment(_, _),
+                out(_, _),
+                out_unescaped(_, _),
+                statement(_, _)),
+        SyntaxTokens = Frontend
+    ) ; ( 
+        must_be(oneof([simple, semblance]), Frontend),
+        named_syntax_tokens(Frontend, SyntaxTokens)
```

Comment:
This could be changed to a search for `named_syntax_tokens(Frontend, SyntaxTokens)` and error on failure to match rather than using `must_be`. Would prevent duplicating the list of valid `Frontend` values.

Reply:
I'm going to update this to only allow valid `syntax_tokens(...)` or `simple` or `semblance` and produce correct error. Right now it's doing it but if the `syntax_tokens` term is not valid, for example, missing the option for `statement` then the predicate will throw wrong error message.